### DISCUSSION TITLE
How to get the stickers / sticker pack by sticker pack ID?

### DISCUSSION THREAD

**Discussion by continue47** [signalstickers/signalstickers#931] [Q&A]
Hi,

Is there a way to retrieve a sticker pack (images, emojis) with just the pack id?

I found signalstickers.org while researching how to get the images / emoji's for stickers based on pack id.

I'm wondering if signalstickers.org has some sort of API or package that I can install to achieve this since I see I can access signalstickers.org/pack/<PACK ID>.
I _could_ just scrape this page by ID, cache it, and get the stickers that way, but there must be a better way (also could be blocked by bot detection, cloudflare, etc).

Any help is appreciated, thanks! :)

**Comment by romainricard** [comment]
[signalstickers-client](https://github.com/signalstickers/signalstickers-client) might help you. But in any case, you'll need the pack key to decrypt the pack. For the packs on signalstickers.org, you can use `https://api.signalstickers.org/v1/packs/`

**Comment by continue47** [comment]
signalstickers-client looks promising. I just ran one of their examples. I'll look more into it to see if I can get the key from it.

https://api.signalstickers.org/v1/packs/ also looks good, but as far as I understand it, sticker packs need to be added there, right? It's not the complete list of sticker packs.

For now I'll keep trying with the python package you provided, thanks for the reply!

**Reply by romainricard** [reply]
You can only get the key if you know it ; packs are encrypted, and are not centralized on Signal's side. That's the whole point of signalstickers.org. And yes, packs on signalstickers.org need to be submited and approved.
